src/pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -370,8 +370,8 @@ def _print_results(results: List[_OUT_DICT], stage: str) -> None:
 
                 try:
                     # some terminals do not support this character
-                    if sys.stdout.encoding is not None:
-                        "─".encode(sys.stdout.encoding)
+                    if sys.__stdout__.encoding is not None:
```

Comment:
Given your motivation, wouldn't it be better to check `hasattr(sys.stdout, "encoding")`?

After all, `sys.stdout` is what we will use to print, and `__stdout__` is just a reference to the original handle: https://docs.python.org/3/library/sys.html#sys.__stdout__

Reply:
`hasattr(sys.stdout, "encoding") and sys.stdout.encoding is None` is an option that prevents an `AttributeError` being raised but for the case where one is writing to a terminal as well as a log file (like in https://stackoverflow.com/a/14906787) you lose the handling of the `─` character.

Reply:
That particular pattern may be actually be rare and in that case I would favour just doing `hasattr(sys.stdout, "encoding") and sys.stdout.encoding is None`.